Common/common.cmake

```diff
@@ -0,0 +1,145 @@
+    ${STEINBERG_DIR}/public.sdk/source/vst2.x/vstplugmain.cpp
+    ${STEINBERG_DIR}/public.sdk/source/vst2.x/aeffeditor.h
+    ${STEINBERG_DIR}/public.sdk/source/vst2.x/audioeffect.h
+    ${STEINBERG_DIR}/public.sdk/source/vst2.x/audioeffectx.h
+  )
+
+  set(VST_INTERFACE
+    ${STEINBERG_DIR}/pluginterfaces/vst2.x/aeffect.h
+    ${STEINBERG_DIR}/pluginterfaces/vst2.x/aeffectx.h
+    ${STEINBERG_DIR}/pluginterfaces/vst2.x/vstfxstore.h
+  )
+
+  source_group("vst2.x" FILES ${VST_SOURCE})
+  source_group("Interfaces" FILES ${VST_INTERFACE})
+  
+  target_sources(${VST_TARGET} PUBLIC ${VST_SOURCE} ${VST_INTERFACE})
+  target_include_directories(${VST_TARGET} PUBLIC ${STEINBERG_DIR})
+  
+endfunction(add_vstsdk)
+
+#*******************************************************************************
+# Adds VSTGUI to target
+#
+# @param VST_TARGET        The cmake target to which the VSTGUI libray will be 
+#                          added.
+# @param VST_TARGET_IMAGES The images used in the cmake targets GUI
+#*******************************************************************************
+function(add_vstgui VST_TARGET VST_TARGET_IMAGES)
+
+  set(VSTGUI_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../vstgui/vstgui)
+
+  set(VSTGUI_SOURCE
+    ${VSTGUI_DIR}/plugin-bindings/aeffguieditor.cpp
+    ${VSTGUI_DIR}/plugin-bindings/aeffguieditor.h
+  )
+
+  IF(WIN32)
+    
+    list(APPEND VSTGUI_SOURCE ${VSTGUI_DIR}/vstgui_win32.cpp)
+    # TODO Generate rc files - for now make sure projects use the same path
+    target_sources(${VST_TARGET} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/resource.rc)
+    
+  ELSEIF(APPLE)
+    
+    list(APPEND VSTGUI_SOURCE ${VSTGUI_DIR}/vstgui_mac.mm)
+    find_library(CARBON Carbon)
+    find_library(COCOA Cocoa)
+    find_library(OPENGL OpenGL)
+    find_library(ACCELERATE Accelerate)
+    find_library(QUARTZ QuartzCore)
+    target_link_libraries(
+      ${VST_TARGET} ${CARBON} ${COCOA} ${OPENGL} ${ACCELERATE} ${QUARTZ}
+    )
+    set_source_files_properties(${VST_TARGET_IMAGES} PROPERTIES 
+      MACOSX_PACKAGE_LOCATION Resources
+    )
+    
+  ENDIF(WIN32)
+  
+  source_group("vstgui" FILES ${VSTGUI_SOURCE})
+  
+  target_sources(${VST_TARGET} PUBLIC ${VSTGUI_SOURCE} ${VST_TARGET_IMAGES})
+  target_include_directories(${VST_TARGET} PUBLIC ${VSTGUI_DIR})
+  
+endfunction(add_vstgui)
+
+#*******************************************************************************
+# Generates a VST cmake target
+#
+# @param VST_TARGET        The name of the target to generate
+# @param VST_TARGET_IMAGES The images used in the targets GUI. If the target 
+#                          doesn't have a GUI then pass FALSE to disable.
+#*******************************************************************************
+function(build_vst VST_TARGET VST_TARGET_SOURCES VST_TARGET_IMAGES)
+  
+  set(COMMON_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../Common)
+  
+  add_library(${VST_TARGET} MODULE ${VST_TARGET_SOURCES})
+  
+  add_vstsdk("${VST_TARGET}")
```

Comment:
As far as I know function calls require parenthesis. 

Unless you meant the double quotes, in which case Cmake requires double quotes to pass the entire contents of a list. Without double quotes it will only pass the first item from a list. This is super ugly, I know. I only figured this out when my builds started failing because they weren't including all the source files. 

In this particular case we probably don't need it because the target should only be a single item but after getting burned I figured it was better safe than sorry.